kernel/src/main/javacc/Outer.jj

```diff
@@ -665,15 +666,29 @@ void Attributes(ASTNode node) : {}
 
 /** Parses a single attribute and adds it to 'node' */
 void Tag(ASTNode node) :
-{ Location loc = startLoc(); String key; String val = "";
-  StringBuilder sb = new StringBuilder(); }
+{ Location loc = startLoc();  }
 {
-  <KEY> { key = image(); }
-  ( "(" { SwitchTo(TAG_STATE); }
-    ( val = String() ")"
-    | TagContent(sb) ")" { val = sb.toString() + special(); } )
-    { SwitchTo(ATTR_STATE); })?
-  { node.addAttribute(key, val); }
+  (
+    { List<String> groups = new ArrayList<String>(); }
+    <GROUP> "(" <KEY> { groups.add(image()); } ("," <KEY> { groups.add(image()); })* ")"
+    {
+      for (String group : groups ) {
+        node.addGroup(group);
+      }
+    }
+  )
+  |
+  (
+    { String key; String val = ""; StringBuilder sb = new StringBuilder(); }
+    <KEY> { key = image(); }
+    ( "(" { SwitchTo(TAG_STATE); }
+      ( val = String() ")"
+      | TagContent(sb) ")" { val = sb.toString() + special(); }
+      )
+      { SwitchTo(ATTR_STATE); }
+    )?
+    { node.addAttribute(key, val); }
+  )
```

Comment:
Huh? Why did you modify the outer parser?
It should be able to just parse `group(my, att)`.
I wouldn't insert more logic into parsing attributes.
If you want to modify what the compiler passes see right after parsing, you can modify this pass: https://github.com/runtimeverification/k/blob/master/kernel/src/main/java/org/kframework/kore/convertors/KILtoKORE.java#L324

Reply:
Not that big of a deal, but I wanted to avoid accepting quoted inputs because they're unnecessary here and their semantics feel a bit odd in this context.

For example, `group("my, foo")` would be parsed the same as `group(my, att)`, but things like `group("my", "att")` aren't accepted. I then also have to do manual error checking for cases like `group(my,,,, att)`.

On the second thought though, I guess we already deal with these same things for attributes like `seqstrict`, so I'll pull the logic into `KILtoKORE` as you suggest.

Reply:
In general, you want to allow your parser to accept more and give better, more informed error messages by analyzing the resulting AST.
If you restrict the language in the parser, you will get _unexpected token x_. And compilation stops.
If you analyze the AST you could say _tags are not allowed to be null_. And as an added bonus, you can return multiple error messages.